pkg/discovery/dtofactory/volume_entity_dto_builder.go

```diff
@@ -103,7 +106,7 @@ func (builder *volumeEntityDTOBuilder) getVolumeCommoditiesSold(vol *api.Persist
 		commBuilder.Capacity(capacity)
 		commodityPerPod, err := commBuilder.Create()
 		if err != nil {
-			return nil, volumeCapacity, err
+			return commoditiesSold, volumeCapacity, err
 		}
 
 		// TODO(irfanurrehman): Set resisable depending on node properties
```

Comment:
Nit: Although it's less likely an error could occur here, would it be better to just skip appending this commoditySold if there's error here and continue on the next podVol? Something like:
```
if err != nil {
    glog.Errorf("Error creating commoditySold by volume %s: %v ", podVol.MountName, err)
} else {
    // TODO(irfanurrehman): Set resisable depending on node properties
    commoditiesSold = append(commoditiesSold, commodityPerPod)
}
```

Reply:
Thanks @marvelyue. Updated!